Library/Homebrew/cmd/reinstall.rb

```diff
@@ -2,6 +2,14 @@
 
 module Homebrew
   def reinstall
+    if !MacOS.can_build?
```

Comment:
This condition is also checked inside `FormulaInstaller#install`; does it need to happen here?